# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/xhr/XMLHttpRequestWorker.cpp
# Commit: a6b021b3776a
# Full Hash: a6b021b3776ab954b0bf56d7a5641b072a719e2f
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2020-06-27 09:38:32
# Regressor Bug: 1645510
# File Overlap Count: 1
# Description:
#   Bug 1645510: Part 3 - Make some UnprivilegedJunkScope calls fallible. r=mccr8
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79721
# ==============================================================================

diff -r 93bca0225002 -r a6b021b3776a dom/xhr/XMLHttpRequestWorker.cpp
--- a/dom/xhr/XMLHttpRequestWorker.cpp	Sat Jun 27 03:06:28 2020 +0000
+++ b/dom/xhr/XMLHttpRequestWorker.cpp	Sat Jun 27 03:06:35 2020 +0000
@@ -1266,9 +1266,8 @@
   if (!mBlobImpl) {
     payload.SetNull();
   } else {
-    AutoSafeJSContext cx;
-
-    JS::Rooted<JSObject*> globalObject(cx, JS::CurrentGlobalOrNull(cx));
+    JS::Rooted<JSObject*> globalObject(RootingCx(),
+                                       xpc::UnprivilegedJunkScope(fallible));
     if (NS_WARN_IF(!globalObject)) {
       aRv.Throw(NS_ERROR_FAILURE);
       return;
